saddle-jmh/src/main/scala/org/saddle/LoopOptimizations.scala

```diff
@@ -0,0 +1,73 @@
+package org.saddle
+
+import org.openjdk.jmh.annotations._
+
+/** Benchmark suite comparing different loop patterns and the impact of loop optimizations.
+  * https://wiki.openjdk.java.net/pages/viewpage.action?pageId=20415918
+  */
+@State(Scope.Benchmark)
+@Warmup(iterations = 10) 
+@Measurement(iterations = 10)
+@Fork(1)
+@Threads(1)
+class LoopOptimisations {
+  @Param(Array("10000"))
+  var size: Int = _
+  var arr: Array[Double] = _
+  var vec1: Vec[Double] = _
+  var b: Double = _
+
+  @Setup(Level.Iteration)
+  def setup() = {
+    arr = vec.rand(size).toArray
+    vec1 = Vec(arr)
+    b = scala.util.Random.nextDouble()
+  }
+
+  @Benchmark
+  /** Simplest variant, should trigger vectorization and other optimisations.
+    */
+  def array(): Array[Double] = {
+    var i = 0
+    while (i < arr.length) {
+      arr(i) = arr(i) + b
```

Comment:
Interestingly, assignment operators seem to harm performance. Here `arr(i) += b` performs worse.